packages/browser/src/Session.ts

```diff
@@ -52,6 +53,28 @@ export interface ISessionOptions {
   clientAuthentication: ClientAuthentication;
 }
 
+async function silentlyAuthenticate(
+  sessionId: string | null,
+  clientAuthn: ClientAuthentication
+) {
+  if (sessionId === null) {
+    // If there is no current session defined, there is no ID token, so silent
+    // authentication cannot happen.
+    return;
+  }
+  // Check if we have an ID Token in storage - if we do then we may be
+  // currently logged in, and the user has refreshed their browser page. In
+  // this case, it can be really useful to save the user's current browser
+  // location, so that we can restore that location after we complete the
+  // entire re-login-the-user-silently-flow (e.g., if the user was on a
+  // specific 'page' of a Single Page App, then presumably they'll expect to
+  // be brought back to exactly that same 'page' after the full refresh).
+  const storedSessionInfo = await clientAuthn.getSessionInfo(sessionId);
+  if (storedSessionInfo?.idToken !== undefined) {
+    window.localStorage.setItem(KEY_CURRENT_URL, window.location.href);
+  }
+}
+
```

Comment:
I think I'd just tweak this to always only work **_only_** with a sessionID, and put the sessionID===null check outside this function.
```suggestion
async function silentlyAuthenticate(
  sessionId: string,
  clientAuthn: ClientAuthentication
) {
  // Check if we have an ID Token in storage - if we do then we may be
  // currently logged in, and the user has refreshed their browser page. In
  // this case, it can be really useful to save the user's current browser
  // location, so that we can restore that location after we complete the
  // entire re-login-the-user-silently-flow (e.g., if the user was on a
  // specific 'page' of a Single Page App, then presumably they'll expect to
  // be brought back to exactly that same 'page' after the full refresh).
  const storedSessionInfo = await clientAuthn.getSessionInfo(sessionId);
  if (storedSessionInfo?.idToken !== undefined) {
    window.localStorage.setItem(KEY_CURRENT_URL, window.location.href);
  }
}
```